lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -89,3 +89,137 @@ class Meta:
             'startDateTime',
             'endDateTime',
         )
+
+
+class LtiAgsScoreSerializer(serializers.ModelSerializer):
+    """
+    LTI AGS LineItemScore Serializer.
+
+    This maps out the internally stored LtiAgsScore to
+    the LTI-AGS API Specification, as shown in the example
+    response below:
+
+    {
+      "timestamp": "2017-04-16T18:54:36.736+00:00",
+      "scoreGiven" : 83,
+      "scoreMaximum" : 100,
+      "comment" : "This is exceptional work.",
+      "activityProgress" : "Completed",
+      "gradingProgress": "FullyGraded",
+      "userId" : "5323497"
+    }
+
+    Reference:
+    https://www.imsglobal.org/spec/lti-ags/v2p0#example-application-vnd-ims-lis-v1-score-json-representation
+    """
+
+    timestamp = serializers.DateTimeField()
+    # All 'scoreGiven' values MUST be positive numeric (including 0).
+    # 'scoreMaximum' represents the denominator and MUST be present when 'scoreGiven' is present
+    scoreGiven = serializers.FloatField(source='score_given', required=False, allow_null=True, default=None)
+    scoreMaximum = serializers.FloatField(source='score_maximum', required=False, allow_null=True, default=None)
+    comment = serializers.CharField(required=False, allow_null=True)
+    activityProgress = serializers.CharField(source='activity_progress')
+    gradingProgress = serializers.CharField(source='grading_progress')
+    userId = serializers.CharField(source='user_id')
+
+    def validate_scoreGiven(self, value):
+        if not self.instance and value is None:
+            raise serializers.ValidationError('Cannot delete score. No score currently exists for this user')
+        return value
```

Comment:
This does not comply to the spec. 
This is a valid score request:
```
{
  "timestamp": "2017-04-16T18:54:36.736+00:00",
  "activityProgress" : "Completed",
  "gradingProgress" : "PendingManual",
  "userId" : "5323497"
}
```
If this is the first request, it'll fail. Remove this validator and make `score_given` allow null values in the model.